### PR TITLE
Strict checking for http-binding package

### PR DESCRIPTION
**Pull request by relu91**
Yet another big PR, really sorry but I couldn't make it shorter because otherwise, the code base would not even compile. 

relates to #758

### CODE REVIEW

File: packages/binding-http/src/credential.ts [hunk 1]
```diff
@@ -167,9 +167,9 @@ export class TuyaCustomBearer extends Credential {
     protected key: string;
     protected secret: string;
     protected baseUri: string;
-    protected token: string;
-    protected refreshToken: string;
-    protected expireTime: Date;
+    protected token: string | undefined;
```
Comment (danielpeintner) [inline packages/binding-http/src/credential.ts:170]: I am just wondering.. would the following not work?

`protected token?: string;`

OR do you want to convey that `token` is **not** optional?

**Review by danielpeintner — APPROVED** [review]
I looked through the code and I did not spot any issue..

Thanks for all the work!

Below a minor question...

File: packages/binding-http/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,8 @@
     "extends": "../../tsconfig.json",
     "compilerOptions": {
         "outDir": "dist",
-        "rootDir": "src"
+        "rootDir": "src",
+        "strict": true
```
Comment (danielpeintner) [inline packages/binding-http/tsconfig.json:6]: Do we want to add 

`"strictFunctionTypes": true`

as we do in the other tsconfigs?

Comment (danielpeintner) [reply packages/binding-http/tsconfig.json:6]: I saw the comment in https://github.com/eclipse-thingweb/node-wot/pull/1047#discussion_r1281537969
Hence we can tackle this comment as addressed

File: packages/binding-http/src/credential.ts (hunk 1 above)
Comment (relu91) [reply packages/binding-http/src/credential.ts]: addressed with https://github.com/eclipse-thingweb/node-wot/pull/1048/commits/5469a3ada4a6da4a368388bc02a4aa94276caf38 (I checked the whole file to have consistency, but in other places we might use the old way)

**Review by danielpeintner — APPROVED** [review]
LGTM

**Review by danielpeintner — APPROVED** [review]

**Pull request merged**